aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/BaseHandlerStd.java

```diff
@@ -30,4 +44,84 @@
     final CallbackContext callbackContext,
     final ProxyClient<CodeartifactClient> proxyClient,
     final Logger logger);
+
+  protected ProgressEvent<ResourceModel, CallbackContext> associateExternalConnections(
+      final ProgressEvent<ResourceModel, CallbackContext> progress,
+      final CallbackContext callbackContext,
+      final ResourceHandlerRequest<ResourceModel> request,
+      final ProxyClient<CodeartifactClient> proxyClient,
+      final Set<String> externalConnectionsToAdd,
+      final Logger logger
+  ) {
+      ResourceModel resourceModel = request.getDesiredResourceState();
+      logger.log(String.format("Adding external connections: %s", externalConnectionsToAdd.toString()));
+      if (CollectionUtils.isNullOrEmpty(externalConnectionsToAdd)) {
+          logger.log("No external connections to add");
+          // Nothing to add, continue
+          return ProgressEvent.progress(resourceModel, callbackContext);
+      }
+
+      if (externalConnectionsToAdd.size() > 1) {
```

Comment:
If I understand this code correctly, this check won't catch if the user first creates the repo with a single external connection, then updates the template to add another external connection. In that case the UpdateHandler does a set difference to see there is one external connection add and calls this associateExternalConnections method with one, so it wouldn't fail this validation.

Why do we even need this check if the service enforces it? And do we really want to have to change this code if were were to make a change to the service to allow it?

Reply:
Yes I agree, it'd be better to decouple this behavior. I'll remove these checks and let the service handle it